### PR TITLE
Feature/improve edgeindex covered

### PR DESCRIPTION
**Pull request by mchacki**
Improve the edge index performance, if filter is based on `_from` and return value is only `_to` because both are stored in the index anyways.

### CODE REVIEW

File: arangod/RocksDBEngine/RocksDBEdgeIndex.cpp [hunk 1]
```diff
@@ -259,18 +259,28 @@ bool RocksDBEdgeIndexIterator::nextCovering(DocumentCallback const& cb, size_t l
   }
 #endif
 
+  VPackBuilder coveringBuilder;
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBEdgeIndex.cpp]: Can we do better here, and reuse an existing temporary Builder from the transaction, by using `TransactionBuilderLeaser` here?

**Comment by mchacki** [comment]
Next attempt on jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2502/

File: CHANGELOG [hunk 2]
```diff
@@ -1,6 +1,50 @@
 devel
 -----
 
+* An AQL query that uses the edge index only and returns the opposite side of
+  the edge can now be executed in a more optimized way:
```
Comment (Simran-B) [inline CHANGELOG]:
```suggestion
  the edge can now be executed in a more optimized way, e.g.
```

File: CHANGELOG [hunk 3]
```diff
@@ -1,6 +1,50 @@
...
+  e.g. FOR edge IN edgeCollection FILTER edge._from == "v/1" RETURN edge._to
```
Comment (Simran-B) [inline CHANGELOG]:
```suggestion

    FOR edge IN edgeCollection FILTER edge._from == "v/1" RETURN edge._to
    
```

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2597/

**Pull request merged**